frontend/public/src/factories/course.js

```diff
@@ -134,3 +146,183 @@ export const makeProgramEnrollment = (): ProgramEnrollment => ({
+): LearnerRecordShare => ({
+  share_uuid:     casual.uuid,
+  created_on:     casual.date(),
+  updated_on:     casual.date(),
+  is_active:      casual.coin_flip,
+  user:           casual.integer(0, 1000),
+  program:        casual.integer(0, 1000),
+  partner_school: forPartnerSchool ? casual.integer(0, 1000) : null
+})
+
+export const makeLearnerRecordUser = (): LearnerRecordUser => ({
+  name:     casual.name,
+  email:    casual.email,
+  username: casual.username
+})
+
+export const makeRequirementCourseNode = (
+  course: LearnerRecordCourse,
+  parentNode: ProgramRequirement,
+  shouldBeCompleted: boolean
+): ProgramRequirement => {
+  const courseNode = {
+    id:   genProgramRequirementId.next().value,
+    data: {
+      node_type: "course",
+      program:   parentNode.data.program,
+      course:    course.id
+    },
+    children: []
+  }
+
+  if (shouldBeCompleted || casual.coin_flip) {
+    course.grade = makeLearnerRecordGrade()
+    course.certificate = makeLearnerRecordCertificate()
+  }
+
+  course.reqtype = parentNode.data.title
+
+  return courseNode
+}
+
+export const makeLearnerRecord = (
+  shouldBeCompleted: boolean
+): LearnerRecord => {
+  // This does a lot of things. It will generate a program with a handful of
+  // courses, some of which are required and some are electives, with the
+  // requisite requirements tree, and will generate grades for your fake learner
+  // depending on the flag.
+
+  const partnerSchools = []
+  const courses = []
+
+  for (let i = 0; i < 5; i++) {
+    partnerSchools.push(makePartnerSchool())
+  }
+
+  // mirroring RC DEDP as of 12/1 - 3 required, two electives and then a nested
+  // elective with two more courses in it
+  for (let i = 0; i < 7; i++) {
+    courses.push(makeLearnerRecordCourse())
+  }
+
+  const learnerRecord = {
+    user:            makeLearnerRecordUser(),
+    program:         makeLearnerRecordProgram(),
+    sharing:         [],
+    partner_schools: partnerSchools
+  }
+
+  // make root node
+  // root nodes have two children - both operators, one for reqs and one for electives
+  const rn = {
+    id:   genProgramRequirementId.next().value,
+    data: {
+      node_type: "program_root",
+      program:   learnerRecord.program.id
+    },
+    children: [
+      {
+        id:   genProgramRequirementId.next().value,
+        data: {
+          node_type: "operator",
+          program:   learnerRecord.program.id,
+          operator:  "all_of",
+          title:     "Required Courses"
+        },
+        children: []
+      },
+      {
+        id:   genProgramRequirementId.next().value,
+        data: {
+          node_type:      "operator",
+          program:        learnerRecord.program.id,
+          operator:       "min_number_of",
+          operator_value: 1,
+          title:          "Elective Courses"
+        },
+        children: []
+      }
+    ]
+  }
+
+  // add courses to the Required Courses node
+  for (let i = 0; i < 3; i++) {
```

Comment:
where the number 3 comes from?

Reply:
DEDP on RC has 3 required courses. Similarly, it has 2 electives and 2 nested electives.